contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpHandler.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.opencensus.common.ExperimentalApi;
+import io.opencensus.contrib.http.util.HttpTraceUtil;
+import io.opencensus.trace.AttributeValue;
+import io.opencensus.trace.MessageEvent;
+import io.opencensus.trace.MessageEvent.Type;
+import io.opencensus.trace.Span;
+import javax.annotation.Nullable;
+
+/**
+ * Base class for handling request on http client and server.
+ *
+ * @param <Q> the HTTP request entity.
+ * @param <P> the HTTP response entity.
+ * @since 0.18
+ */
+@ExperimentalApi
+public abstract class HttpHandler<Q, P> {
```

Comment:
Why is the class public? Probably it can be package protected. Also maybe consider to rename it to "AbstractHttpHandler" or "BaseHttpHandler".

Reply:
done.